### PR TITLE
PPS: backport of #28492, #28615 and #28621

### PR DESCRIPTION
**Pull request by jan-kaspar**
This is a cumulative backport of #28492, #28615 and #28621.

This PR does not change the results of default proton reconstruction. The plot below compares reco proton kinematic distributions from CMSSW_10_6_8 (blue) and this PR (red dashed). For xi histograms the perfect overlap is obvious. The apparent difference in the other histograms to due to a different binning.

make_cmp

This PR provides the same "direct" simulation as in master. The plot below compares a simulated x distribution from master (blue) and this PR (red dashed) - there is perfect overlap.

x_dist_cmp

### CODE REVIEW

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
@jan-kaspar 
please describe the reasons for the backport.
Is this needed for any parts of the ultra-legacy production (reco or re-miniAOD)?

**Comment by jan-kaspar** [comment]
> @jan-kaspar
> please describe the reasons for the backport.
> Is this needed for any parts of the ultra-legacy production (reco or re-miniAOD)?

Not directly. This PR essentially provides an update of the ["direct simulation"](https://twiki.cern.ch/twiki/bin/viewauth/CMS/TaggedProtonsDirectSimulation). Since this is the only simulation usable by PPS, people have started to use it for physics analyses. And for that reason they need to have it in a CMSSW version compatible with the UL re-reco.

**Comment by jfernan2** [comment]
+1
Changes in Validation/CTPSS not manifested in the DQM comparison since CT-PPS Validation is not enabled yet

**Comment by tlampen** [comment]
+1

File: Geometry/VeryForwardGeometryBuilder/src/DetGeomDesc.cc [hunk 1]
```diff
@@ -27,8 +28,14 @@ DetGeomDesc::DetGeomDesc( DDFilteredView* fv ) :
   m_name((( fv->logicalPart()).ddname()).name()),
   m_params((( fv->logicalPart()).solid()).parameters()),
   m_copy( fv->copyno()),
-  m_z( fv->geoHistory().back().absTranslation().z())
-{}
+  m_z( fv->geoHistory().back().absTranslation().z()),
```
Comment (ianna) [inline Geometry/VeryForwardGeometryBuilder/src/DetGeomDesc.cc]: @jan-kaspar - please, move access to **DDFilteredView** out of the constructor and provide all these parameters in an explicit **DetGeomDesc** constructor.

**Review by ianna — CHANGES_REQUESTED** [review]

File: Geometry/VeryForwardGeometryBuilder/src/DetGeomDesc.cc (hunk 1 above)
Comment (civanch) [reply Geometry/VeryForwardGeometryBuilder/src/DetGeomDesc.cc]: @ianna , this is the back-port to legacy Run-2 release. Your request is fully valid to the current master software but not for back-porting of already merged PRs.

Comment (jan-kaspar) [reply Geometry/VeryForwardGeometryBuilder/src/DetGeomDesc.cc]: Several updates (in master) to PPS geometry are [planned](https://docs.google.com/spreadsheets/d/1Yv_2uDdqvFGLTjspm60fnUMQfLr3geuNF_xXfpd_uWI/edit?usp=sharing) with ETA by March.

Comment (ianna) [reply Geometry/VeryForwardGeometryBuilder/src/DetGeomDesc.cc]: > Several updates (in master) to PPS geometry are [planned](https://docs.google.com/spreadsheets/d/1Yv_2uDdqvFGLTjspm60fnUMQfLr3geuNF_xXfpd_uWI/edit?usp=sharing) with ETA by March.

@cvuosalo - FYI

**Comment by ianna** [comment]
+1

**Comment by perrotta** [comment]
+1
- Mostly Validation: the reco part of this PR is correctly backported
- Jenkins tests show no differences, as expected because this PR does not change the results of default proton reconstruction.

**Comment by jan-kaspar** [comment]
@alberto-sanchez, @agrohsje, @efeyazgan, @mkirsano, @qliphy, @SiewYan : May I please ask for review? This PR only adds one new _cfi.py file in Configuration/Generator. Thanks in advance!

**Comment by jan-kaspar** [comment]
@ggovi : May I please ask for review? The changes in CondFormats are small and technical and have already been signed off for master: https://github.com/cms-sw/cmssw/pull/28492#issuecomment-562032489
Thanks in advance!

**Comment by ggovi** [comment]
+1

**Comment by silviodonato** [comment]
@alberto-sanchez, @agrohsje, @efeyazgan, @mkirsano, @qliphy, @SiewYan this PR simply add a configuration file in Configuration/Generator. Could you please review/sign it?

**Comment by perrotta** [comment]
I'm curious: why the generator signature is needed here?

**Comment by perrotta** [comment]
> I'm curious: why the generator signature is needed here?

@alberto-sanchez,@SiewYan,@qliphy,@efeyazgan,@mkirsano,@agrohsje please have a look and sign, if you think so

**Comment by silviodonato** [comment]
@perrotta It is a minor change. They are moving Validation/CTPPS/python/year_2016/randomXiThetaGunProducer_cfi.py → Configuration/Generator/python/randomXiThetaGunProducer_cfi.py .

**Comment by silviodonato** [comment]
code-checks

**Comment by silviodonato** [comment]
merge
@alberto-sanchez,@SiewYan,@qliphy,@efeyazgan,@mkirsano,@agrohsje if you have any comments, please let us know

**Pull request merged**